### PR TITLE
Fix violating consider-using-set-comprehension pylint rule

### LINKED ISSUE #6531: Enable `consider-using-set-comprehension` for pylint
Enable the `consider-using-set-comprehension` rule (https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-set-comprehension.html) to improve readability and avoid transient list creation.

Instructions:

1. Add `consider-using-set-comprehension` in `pylintrc`:

```diff
diff --git a/pylintrc b/pylintrc
index 24ff2174e..a86adc713 100644
--- a/pylintrc
+++ b/pylintrc
@@ -179,7 +179,8 @@ enable=c-extension-no-member,
        consider-using-from-import,
        consider-merging-isinstance,
        consider-using-dict-items,
-       unnecessary-comprehension
+       unnecessary-comprehension,
+       consider-using-set-comprehension
```

2. Run `pylint $(git ls-files | grep '\.py$')`. The output should look like:

```
************* Module mlflow.pipelines.utils.metrics
mlflow/pipelines/utils/metrics.py:58:26: R1718: Consider using a set comprehension (consider-using-set-comprehension)
mlflow/pipelines/utils/metrics.py:59:27: R1718: Consider using a set comprehension (consider-using-set-comprehension)
************* Module mlflow.store.model_registry.sqlalchemy_store
mlflow/store/model_registry/sqlalchemy_store.py:553:34: R1718: Consider using a set comprehension (consider-using-set-comprehension)
mlflow/store/model_registry/sqlalchemy_store.py:555:34: R1718: Consider using a set comprehension (consider-using-set-comprehension)
...
```

3. Fix the lines listed in the previous step.
4. File a PR.

### PR DESCRIPTION
**Pull request by iamthen0ise**
## Related Issues/PRs
Closes #6531 

## What changes are proposed in this pull request?

Enables `consider-using-set-comprehension` pylint rule and fixes violations.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 1]
```diff
@@ -337,8 +337,8 @@ def test_list_experiments_paginated_last_page(self):
             returned_experiments.extend(result)
         self.assertEqual(result.token, None)
         # make sure that at least all the experiments created in this test are found
-        returned_exp_id_set = set([exp.experiment_id for exp in returned_experiments])
-        self.assertEqual(set(experiments) - returned_exp_id_set, set())
+        returned_exp_id_set = {exp.experiment_id for exp in returned_experiments}
+        self.assertEqual({experiments} - returned_exp_id_set, set())
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:341]:
```suggestion
        self.assertEqual(set(experiments) - returned_exp_id_set, set())
```

`{x}` is not equivalent to `set(x)`.

File: tests/store/tracking/__init__.py [hunk 2]
```diff
@@ -61,8 +61,8 @@ def _verify_logged(store, run_id, metrics, params, tags):
         all_metrics = sum([store.get_metric_history(run_id, key) for key in run.data.metrics], [])
         assert len(all_metrics) == len(metrics)
         logged_metrics = [(m.key, m.value, m.timestamp, m.step) for m in all_metrics]
-        assert set(logged_metrics) == set([(m.key, m.value, m.timestamp, m.step) for m in metrics])
-        logged_tags = set(run.data.tags.items())
-        assert set([(tag.key, tag.value) for tag in tags]) <= logged_tags
+        assert {logged_metrics} == {(m.key, m.value, m.timestamp, m.step) for m in metrics}
```
Comment (harupy) [inline tests/store/tracking/__init__.py:64]:
```suggestion
        assert set(logged_metrics) == {(m.key, m.value, m.timestamp, m.step) for m in metrics}
```

File: tests/store/tracking/__init__.py [hunk 3]
```diff
@@ -61,8 +61,8 @@ def _verify_logged(store, run_id, metrics, params, tags):
...
+        logged_tags = {run.data.tags.items()}
+        assert {(tag.key, tag.value) for tag in tags} <= logged_tags
         assert len(run.data.params) == len(params)
-        assert set(run.data.params.items()) == set([(param.key, param.value) for param in params])
+        assert {run.data.params.items()} == {(param.key, param.value) for param in params}
```
Comment (harupy) [inline tests/store/tracking/__init__.py:68]:
```suggestion
        assert set(run.data.params.items()) == {(param.key, param.value) for param in params}
```

File: tests/store/tracking/__init__.py [hunk 4]
```diff
@@ -61,8 +61,8 @@ def _verify_logged(store, run_id, metrics, params, tags):
         all_metrics = sum([store.get_metric_history(run_id, key) for key in run.data.metrics], [])
         assert len(all_metrics) == len(metrics)
         logged_metrics = [(m.key, m.value, m.timestamp, m.step) for m in all_metrics]
-        assert set(logged_metrics) == set([(m.key, m.value, m.timestamp, m.step) for m in metrics])
-        logged_tags = set(run.data.tags.items())
-        assert set([(tag.key, tag.value) for tag in tags]) <= logged_tags
+        assert {logged_metrics} == {(m.key, m.value, m.timestamp, m.step) for m in metrics}
+        logged_tags = {run.data.tags.items()}
```
Comment (harupy) [inline tests/store/tracking/__init__.py:65]:
```suggestion
        logged_tags = set(run.data.tags.items())
```

**Comment by harupy** [comment]
@iamthen0ise Thanks for the PR! Left some comments.

**Comment by harupy** [comment]
@iamthen0ise Can you fix the DCO check failure (https://github.com/mlflow/mlflow/pull/6540/checks?check_run_id=7968022048)?

**Comment by iamthen0ise** [comment]
looks like it's time to get into the habit of signing commits. Thanks for your help and patience!

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@iamthen0ise Would you be interested in working on https://github.com/mlflow/mlflow/issues/6535?

**Comment by harupy** [comment]
@iamthen0ise Merged! Thanks for the contribution!

**Pull request merged**